### PR TITLE
[New Rule] PE through Writable Docker Socket

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
This rule monitors for the usage of Docker runtime sockets to escalate privileges on Linux systems. Docker sockets by default are only be writable by the root user and docker group. Attackers that have permissions to write to these sockets may be able to create and run a container that allows them to escalate privileges and gain further access onto the host file system.

## Detection
0 hits last 365 days in red sector, nor in my docker test clusters. Added `and not user.Ext.real.id : "0" and not group.Ext.real.id : "0"` to minimize FPs. 
```
process where host.os.type == "linux" and event.type == "start" and event.action == "exec" and 
(
  (process.name == "docker" and process.args : "run" and process.args : "-it"  and 
   process.args : ("unix://*/docker.sock", "unix://*/dockershim.sock")) or 
  (process.name == "socat" and process.args : ("UNIX-CONNECT:*/docker.sock", "UNIX-CONNECT:*/dockershim.sock"))
) and not user.Ext.real.id : "0" and not group.Ext.real.id : "0"
```
image

### CODE REVIEW

File: rules/linux/privilege_escalation_writable_docker_socket.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Privilege Escalation through Writable Docker Socket"
+references = ["https://book.hacktricks.xyz/linux-hardening/privilege-escalation/docker-security/docker-breakout-privilege-escalation#automatic-enumeration-and-escape"]
+risk_score = 47
+rule_id = "7acb2de3-8465-472a-8d9c-ccd7b73d0ed8"
+severity = "medium"
+tags = ["Domain: Endpoint", "OS: Linux", "Use Case: Threat Detection", "Tactic: Privilege Escalation"]
```
Comment (w0rk3r) [inline rules/linux/privilege_escalation_writable_docker_socket.toml:26]:
```suggestion
tags = ["Domain: Endpoint", "OS: Linux", "Use Case: Threat Detection", "Tactic: Privilege Escalation", "Domain: Container"]
```

Comment (w0rk3r) [reply rules/linux/privilege_escalation_writable_docker_socket.toml:26]: This one is an optional nit, not sure if relevant

**Review by w0rk3r — APPROVED** [review]
🔥

**Review by eric-forte-elastic — APPROVED** [review]
Great rule :fire: 

There are some instances where users will need to use docker runtime sockets to route control of the docker interface in some more intricate development environments. However, these are few and far between and I have not seen one that also makes use of `-it` as one might use for privilege escalation. Great Work!

**Pull request merged**